### ISSUE TITLE
Phinx can create tables in multiple schemas, but not modify them!

### ISSUE THREAD

**Issue by joshribakoff** [cakephp/phinx#229]
setup the schemas:

```
mysql> create database development_db;
Query OK, 1 row affected (0.01 sec)

mysql> create database production_db;                                                                                                                                                                        
Query OK, 1 row affected (0.01 sec)
```

phinx.yml

```
paths:
    migrations: %%PHINX_CONFIG_DIR%%/migrations

environments:
    default_migration_table: phinxlog
    default_database: development
    production:
        adapter: mysql
        host: localhost
        name: production_db
        user: root
        pass: ''
        port: 3306

    development:
        adapter: mysql
        host: localhost
        name: development_db
        user: root
        pass: ''
        port: 3306
```

First migration:

``` php
public function change()
    {
        $this->table('production_db.foo')
            ->addColumn('foo', 'string')
            ->save();
    }
```

Outputs:

```
using database development_db

 == 20140417164540 MyTest: migrating
 == 20140417164540 MyTest: migrated 0.0901s

All Done. Took 0.1786s
```

Second migration:

``` php
public function change()
    {
        $this->table('production_db.foo')
            ->addColumn('foo2', 'string')
            ->save();
    }
```

Outputs:

```
using database development_db

 == 20140417164714 MyTest2: migrating



  [PDOException]                                                                       
  SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'foo' already exists  
```

I am then left, of course, with these tables:

```
mysql> show tables in development_db;
+--------------------------+
| Tables_in_development_db |
+--------------------------+
| phinxlog                 |
+--------------------------+
1 row in set (0.00 sec)

mysql> show tables in production_db;
+-------------------------+
| Tables_in_production_db |
+-------------------------+
| foo                     |
+-------------------------+
1 row in set (0.00 sec)

mysql> describe production_db.foo;
+-------+--------------+------+-----+---------+----------------+
| Field | Type         | Null | Key | Default | Extra          |
+-------+--------------+------+-----+---------+----------------+
| id    | int(11)      | NO   | PRI | NULL    | auto_increment |
| foo   | varchar(255) | NO   |     | NULL    |                |
+-------+--------------+------+-----+---------+----------------+
2 rows in set (0.17 sec)
```

TL;DR  = when checking if a table already exists, it uses the "active" database, ignoring the fact that I specified a schema in my table name. It must be checking "development_db" for a table "foo", which doesn't exist, which of course causes it to run a create statement on "production_db" for a table "foo", which does exist, therefore the create statement fails.

One solution could be to abolish checking if tables exist, and use "create if not exists". Or I could work around this in userland, I guess, by passing raw SQL in the `up()` & `down()` and not using the phinx API.
